lib/rubycritic/generators/global_rating_calculator.rb

```diff
@@ -0,0 +1,15 @@
+module Rubycritic
+  module Generator
+    module GlobalRatingCalculator
+      def calculate_gpa(analysed_modules)
+        gpa_sum = 0
+        total_lines = 0
+        analysed_modules.each do |analysed_module|
+          gpa_sum += (analysed_module.rating.to_gpa) * analysed_module.lines
```

Comment:
What's the reasoning for converting the rating back to a GPA step? I think I like #35's approach better, averaging the costs of the modules and then using the Rating class to convert it to a GPA.